### PR TITLE
Incorporate group parent child relationships into variable assignment hierarchy

### PR DESCRIPTION
**Pull request by rbramwell**
**TL;DR:**

Makes variables that are set at the parent group level available to child groups.

**Overview:**

The current code base does away with [parentgroup:children] groups in favour of converting the parent group into a group at the same level as its children. However, this dismisses the fact that parent-child relationships add a level of precedence to variable assignments for groups. This PR adds this logic to better align ansible_spec variable assignments with how this is done in Ansible.

**Example:**

hosts file:

screenshot at 2018-08-22 05-25-50

Makes variables set in ./group_vars/parentgroup/vars (or any files in the dir) available to hosts in the [group1] and [group2] groups.

### CODE REVIEW

File: lib/ansible_spec/load_ansible.rb [hunk 1]
```diff
@@ -77,7 +87,16 @@ def self.load_targets(file)
...
       end
     }
-    return groups
+
+    if return_type == 'groups'
+      return_value = groups
+    elsif return_type == 'groups_parent_child_relationships'
+      return_value = groups_parent_child_relationships
+    else
+      return_value = groups
```
[the remark this answers is not in the document]
Comment (rbramwell) [reply lib/ansible_spec/load_ansible.rb]: That is the default; implicit tests for this already exist. However, I will add an explicit test.

File: lib/ansible_spec/load_ansible.rb [hunk 2]
```diff
@@ -8,9 +8,15 @@
...
 module AnsibleSpec
   # param: inventory file of Ansible
+  # param: return_type 'groups' or 'groups_parent_child_relationships'
   # return: Hash {"group" => ["192.168.0.1","192.168.0.2"]}
   # return: Hash {"group" => [{"name" => "192.168.0.1","uri" => "192.168.0.1", "port" => 22},...]}
-  def self.load_targets(file)
+  # return: Hash {"pg" => ["server", "databases"]}
+  def self.load_targets(file, return_type = 'groups')
+    if not ['groups', 'groups_parent_child_relationships'].include?(return_type)
+      raise ArgumentError, "Variable return_type must be value 'groups' or 'groups_parent_child_relationships'"
```
[the remark this answers is not in the document]
Comment (rbramwell) [reply lib/ansible_spec/load_ansible.rb]: Sure. Will add and update now.

[the remark this answers is not in the document]
Comment (rbramwell) [reply lib/ansible_spec/load_ansible.rb]: Test added and tests passed.

File: lib/ansible_spec/load_ansible.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (rbramwell) [reply lib/ansible_spec/load_ansible.rb]: Tests added and tests passed.

File: lib/ansible_spec/load_ansible.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (rbramwell) [reply lib/ansible_spec/load_ansible.rb]: See https://github.com/volanja/ansible_spec/pull/118/files#diff-1d66ff5cbe33a20ab89e9cf334f001b1R403

File: lib/ansible_spec/load_ansible.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (rbramwell) [reply lib/ansible_spec/load_ansible.rb]: See:

File: lib/ansible_spec/load_ansible.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (rbramwell) [reply lib/ansible_spec/load_ansible.rb]: @volanja Handing back to you for review.

File: lib/ansible_spec/load_ansible.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (rbramwell) [reply lib/ansible_spec/load_ansible.rb]: @volanja Handing back to you for review.

File: lib/ansible_spec/load_ansible.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (volanja) [reply lib/ansible_spec/load_ansible.rb]: @rbramwell Thank you for your changes.

File: lib/ansible_spec/load_ansible.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (volanja) [reply lib/ansible_spec/load_ansible.rb]: @rbramwell Thank you for your changes.
I think that The default value is easier to understand if first defined.
I will modify it after merge this PR.

**Comment by volanja** [comment]
I modified code (e9395816a6dcb994c716c3e8df191f3392284ef8).
Thank you for your contribution!

File: lib/ansible_spec/load_ansible.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (rbramwell) [reply lib/ansible_spec/load_ansible.rb]: Yeah, I'm fine with that.

**Comment by rbramwell** [comment]
Glad to have contributed!
